docs/ARCHITECHTURE.md

```diff
@@ -0,0 +1,46 @@
+# Batarangle Architecture
+
+The architecture of the Batarangle dev tools extensions can be broken down into 3 parts:
+
+1. **Front-end**: display of block/tree view of the inspected application and other debugging information.
```

Comment:
I am not too keen on "front end", "back end", "middleware" terminology, since we are using each of the three words in a rather different way from how they are normally used.